### PR TITLE
[INFRA/CORE] Cleanup

### PR DESCRIPTION
**Pull request by lhl2617**
# Summary

- Tidy up `BaseClient` interface to follow sequence of method definitions
- Remove unused `BaseClient` method implementations
- Remove `CompareTestIntegers`
- Remove `ShortHash` from `GitInfo` -- 7 is an arbitrary number chosen by GitHub.

## Test Plan

Unit tests pass.

### CODE REVIEW

**Comment by QFSW** [comment]
# ⚠️Warning! This diff contains changes in protected folders

File: website/src/containers/AppLayout/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const AppNavbar = () => {
             </Link>
 
             <a rel="noopener noreferrer" target="_blank" href={outputJSONData.GitInfo.GithubURL} className="lightbluelink">
-                {outputJSONData.GitInfo.ShortHash}
+                {outputJSONData.GitInfo.Hash.substr(0, 7)}
```
Comment (QFSW) [inline website/src/containers/AppLayout/Navbar/Navbar.tsx:39]: 0,7? Magic numbers?

**Review by QFSW — APPROVED** [review]
Looks mostly good to me

File: website/src/containers/AppLayout/Navbar/Navbar.tsx (hunk 1 above)
Comment (lhl2617) [reply website/src/containers/AppLayout/Navbar/Navbar.tsx:39]: image

yeah I trust github devs

**Pull request merged**